packages-exp/auth-exp/src/model/auth.d.ts

```diff
@@ -15,12 +15,21 @@
  * limitations under the License.
  */
 
+import {
+  CompleteFn,
+  ErrorFn,
+  NextFn,
+  Observer,
+  Unsubscribe
+} from '@firebase/util';
+
 import { Persistence } from '../core/persistence';
 import { User } from './user';
 
 export type AppName = string;
 export type ApiKey = string;
 export type AuthDomain = string;
+export type NextOrObserver<T> = NextFn<T | null> | Observer<T | null>;
 
 export interface Config {
```

Comment:
What is the `packages-exp/auth-types-exp` package for? I thought it was for the type definitions that users interact with (like this).

Reply:
<product>-types is for public types intended for clients, the types we have defined internally may be more permissive than what we expose to customers since there's no 'protected' keyword in TS.  Some fields that may exist/be writeable in auth-exp, may either not exist or be readonly in auth-type-exp